### PR TITLE
added sqlite3 support

### PR DESCRIPTION
**Pull request by fabianbaier**
Run 
```
go run cmd/tsdb/main.go dump ../../snapshots/snapshots/20191017T205052Z-32c83bebdee150eb --label-name=__name__ --label-value=etcd_disk_wal_fsync_duration_seconds_sum --format=sqlite3
```

### CODE REVIEW

File: tsdb/cmd/tsdb/main.go [hunk 1]
```diff
@@ -64,9 +66,13 @@ func execute() (err error) {
 		analyzeBlockID       = analyzeCmd.Arg("block id", "block to analyze (default is the last block)").String()
 		analyzeLimit         = analyzeCmd.Flag("limit", "how many items to show in each list").Default("20").Int()
 		dumpCmd              = cli.Command("dump", "dump samples from a TSDB")
+		dumpFormat           = dumpCmd.Flag("format", "output format").Default("stdout").String()
 		dumpPath             = dumpCmd.Arg("db path", "database path (default is "+defaultDBPath+")").Default(defaultDBPath).String()
 		dumpMinTime          = dumpCmd.Flag("min-time", "minimum timestamp to dump").Default(strconv.FormatInt(math.MinInt64, 10)).Int64()
 		dumpMaxTime          = dumpCmd.Flag("max-time", "maximum timestamp to dump").Default(strconv.FormatInt(math.MaxInt64, 10)).Int64()
+		dumpLabelName        = dumpCmd.Flag("label-name", "label name").String()
+		dumpLabelValue       = dumpCmd.Flag("label-value", "label value").String()
+		dumpSourceName       = dumpCmd.Flag("source-name", "path to sqlite3 db file").Default("./sqlite3.db").String()
```
Comment (fabianbaier) [inline tsdb/cmd/tsdb/main.go]: source-name is probably the wrong name, I think it should be more `dump-output` or so.

Comment (guenter) [reply tsdb/cmd/tsdb/main.go]: We might support other output formats in the future so I would call it `output-file` or similar

File: tsdb/cmd/tsdb/main.go [hunk 2]
```diff
@@ -617,9 +633,34 @@ func analyzeBlock(b tsdb.BlockReader, limit int) error {
 	return nil
 }
 
-func dumpSamples(db *tsdb.DBReadOnly, mint, maxt int64) (err error) {
+type dumpConfiguration struct {
```
Comment (guenter) [inline tsdb/cmd/tsdb/main.go:636]: what's the convention for placing typedefs and functions? I would expects typedefs at the top of the file like in C, not sure what's expected in Go

File: tsdb/cmd/tsdb/main.go [hunk 3]
```diff
@@ -653,3 +702,107 @@ func dumpSamples(db *tsdb.DBReadOnly, mint, maxt int64) (err error) {
...
+		return fmt.Errorf("error creating data table: %v", err)
+	}
+
+	tx, err := dbSqlite3.Begin()
+	if err != nil {
+		return fmt.Errorf("error beginning transaction: %v", err)
+	}
+	stmt, err := tx.Prepare("INSERT INTO labels(labels) values(?)")
+	if err != nil {
+		return fmt.Errorf("error preparing transaction: %v", err)
```
Comment (guenter) [inline tsdb/cmd/tsdb/main.go]: should be "error preparing statement", also see below for more instances

File: tsdb/cmd/tsdb/main.go [hunk 4]
```diff
@@ -653,3 +702,107 @@ func dumpSamples(db *tsdb.DBReadOnly, mint, maxt int64) (err error) {
...
+	}
+	defer stmt.Close()
+
+	stmtData, err := tx.Prepare("INSERT INTO data(label_id, timestamp, value) values((SELECT id FROM labels WHERE labels = ?), ?, ?);")
```
Comment (guenter) [inline tsdb/cmd/tsdb/main.go:749]: this subquery is probably going to be slow. But you can merge this, we can optimize later if it's too slow.

**Review by guenter — CHANGES_REQUESTED** [review]

File: tsdb/cmd/tsdb/main.go (hunk 2 above)
Comment (fabianbaier) [reply tsdb/cmd/tsdb/main.go:636]: Yeah its a bit different to C in that usually the convention is to keep types close to where they are used. That way its easy for any maintainer to find the types. Now, some core types are typically grouped at the top of a file, but for me `dumpConfiguration` is not so much as a core file as `Configuration` would be. You find a similar pattern with the `writeBenchmark` struct on L159.

**Pull request merged**